mkt/feed/views.py

```diff
@@ -811,6 +815,35 @@ def get_es_feed_element_query(self, sq, feed_items):
 
         return sq.filter(es_filter.Bool(should=filters))[0:len(feed_items)]
 
+    def _featured_website_query(self, region=None, n=11):
+        """
+        Query for n featured websites matching an optional region.
+        """
+        TAG = 'featured-website'
+        if region:
+            TAG = '%s-%s' % (TAG, region.slug)
+        daily_seed = int(datetime.now().strftime('%Y%m%d'))
+        mow_query = query.Q(
+            'function_score',
+            filter=es_filter.Bool(should=[es_filter.Term(tags=TAG)]),
+            functions=[SF('random_score', seed=daily_seed)],
+        )
+        es = Search(using=WebsiteIndexer.get_es())[:n]
+        return es.query(mow_query).execute().hits
+
+    def get_featured_websites(self, n):
+        """
+        Get up to n featured MOWs for the request's region.
+
+        If less than n are available, make up the difference with globally-
+        featured MOWs.
+        """
+        featured = list(self._featured_website_query(region=self.request.REGION))
+        if len(featured) < n:
+            diff = n - len(featured)
+            featured += list(self._featured_website_query(n=diff))
```

Comment:
That's exactly what I tried to do, but it didn't act predictably. I'm going to try it again with a fresh brain, because I don't see why it wouldn't work.